cmd/server/api/rpchandler_test.go

```diff
@@ -0,0 +1,142 @@
+	"go.uber.org/mock/gomock"
+)
+
+func TestPostRpcHandler(t *testing.T) {
+	t.Parallel() // Run tests in parallel
+
+	tests := []struct {
+		name            string
+		requestBody     string
+		requestHeaders  map[string]string
+		expectedHeaders map[string]string
+		mockResponse    string
+		mockError       error
+		expectedStatus  int
+		expectedBody    string
+	}{
+		{
+			name:           "successful request",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+			mockResponse:   `{"jsonrpc":"2.0","result":"1001","id":67}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","result":"1001","id":67}`,
+		},
+		{
+			name: "successful eth_call request with params and headers",
+			requestBody: `{
+        "jsonrpc": "2.0",
+        "method": "eth_call",
+        "params": [{
+            "to": "0xc4d9faef49ec1e604a76ee78bc992abadaa29527",
+            "data": "0xc87b56dd0000000000000000000000000000000000000000000000000000000000000000"
+        }, "latest"],
+        "id": 1
+    }`,
+			requestHeaders:  map[string]string{"X-Custom-Header": "custom_value"},
+			expectedHeaders: map[string]string{"X-Custom-Header": "custom_value"},
+			mockResponse:    `{"jsonrpc":"2.0","id":1,"result":"0x00477777730000000000"}`,
+			expectedStatus:  http.StatusOK,
+			expectedBody:    `{"jsonrpc":"2.0","id":1,"result":"0x00477777730000000000"}`,
+		},
+		{
+			name:           "non successful request",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+			mockResponse:   `{"jsonrpc":"2.0","error":{"code":-32601,"message":"The method net_version does not exist/is not available"}}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","error":{"code":-32601,"message":"The method net_version does not exist/is not available"}}`,
+		},
+		{
+			name:           "non successful request with invalid json",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67`,
+			mockResponse:   `{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"}}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"}}`,
+		},
+		{
+			name:           "client error",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+			mockError:      errors.New("client error"),
+			expectedStatus: http.StatusBadGateway,
+			expectedBody:   "Bad Gateway\n",
+		},
+	}
+
+	for _, ttest := range tests {
+		tt := ttest // Shadow loop variable otherwise it could be overwrittens
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			t.Cleanup(func() {
+				ctrl.Finish()
+			})
+
+			mockHttpClient := mock.NewMockHttpClientInterface(ctrl)
+			handler := api.NewHandler(
+				"https://polygon-mumbai.g.alchemy.com/",
+				api.WithHttpClient(mockHttpClient),
+			)
+			t.Parallel() // Run tests in parallel
```

Comment:
I have two more requests: move this as a first instruction of the sub-test and add `t.Parallel()` to the rest of the tests you've added in this PR, please